src/smdba/postgresqlgate.py

```diff
@@ -150,7 +150,7 @@ def estimate(self) -> "PgTune":
 
         self.config['shared_buffers'] = self.to_mb(self.bin_rnd(mem / 4))
         self.config['effective_cache_size'] = self.to_mb(self.bin_rnd(mem * 3 / 4))
-        self.config['work_mem'] = self.to_mb(self.bin_rnd(mem / self.max_connections))
+        self.config['work_mem'] = self.to_mb(self.bin_rnd(mem - self.config['shared_buffers'] / 3 * self.max_connections))
```

Comment:
```suggestion
        self.config['work_mem'] = self.to_mb(self.bin_rnd((mem - self.config['shared_buffers']) / 3 * self.max_connections))
```

Not sure if we need this extra braces. Please check.

Reply:
Not only there but also around the divisor, see the formula in https://github.com/SUSE/spacewalk/issues/9015. Moreover, `mem` is in bytes while `shared_buffers` is in mebibytes. Actually it's a string, so I'd expect this to not even pass type check.

Please do double check all calculations here, and test them. It might be worth it having a unit test as well.